src/framework/jobState/LocalStepJobState/LocalStepJobState.ts

```diff
@@ -0,0 +1,97 @@
+import { Promisable } from 'type-fest';
+
+import { Entity, Relationship } from '../../types';
+import { JobState, GraphObjectFilter, GraphObjectIteratee } from '../types';
+
+import { flushDataToDisk } from './flushDataToDisk';
+
+export const GRAPH_OBJECT_BUFFER_THRESHOLD = 500; // arbitrarily selected, subject to tuning
+
+interface LocalStepJobStateInput {
+  step: string;
+  cacheDirectory?: string;
+}
+
+export class LocalStepJobState implements JobState {
```

Comment:
I wondered about this name and forgot in the review... What is the reason for including the word "Step"? I don't see how this is limited to step state. Also, what do you think of the name `FileSystemJobState`?

Reply:
That's a much better name than `LocalStepJobState`. @philidem would be proud.

I included `step` in the name because the intent was to create a job state object with reference to the step per execution so that the job state knows where to place data. The more I think about it though, the more it seems like having an individual job state per step is a bad idea considering lots of steps could run in parallel. With how things are implemented here, each state object has it's own set buffer of entities to manage.

I think I'm going to tweak the implementation a bit to allow for `child` job states to be created (kinda like loggers) with context about steps and where to write data, but a shared pool of entities and relationships.

Reply:
Actually... might be better to handle this in a later PR. I have a solution in mind that probs won't take long to implement, but time might be better spent getting the integration sdk in a usable state before Monday. With how small these upcoming integrations will be, I don't think memory will be an issue for a while.

Reply:
Waiting sounds good to me! The most important thing in my mind right now is making it work, clear interfaces/naming, and where to place responsibilities. Performance improvements should definitely come later.

> With how things are implemented here, each state object has it's own set buffer of entities to manage.

I'll admit that I didn't pick up on the idea that each step would have it's own state obj. It *does* make sense that each step has state, but I've been looking at this code so far as the collector of entities and relationships, which probably should cross steps.

This suggests to me that there is likely to be 1) step state and 2) entities/relationships collected across steps. I think it may lead to some confusion to have those concepts mixed into a single thing. Perhaps there is a `FileSystemGraphObjectCollector`, and a separate `StepState` concept having a containment hierarchy reflecting the step dependency graph. 🤔

We'll see! Hopefully those things will be transparent to the integration author and we can iterate the internals. 💪

Reply:
Or `FileSystemGraphObjectStore` 😄